erdos/graph.py

```diff
@@ -143,6 +143,8 @@ def _build_refined_op_graph(self):
                     output_streams = op_handle.op_cls.setup_streams(
                         DataStreams(current_input_streams),
                         **op_handle.setup_args)
+                    for stream in output_streams:
+                        stream.set_uid(op_id)
```

Comment:
Can you please rename to `set_publisher_id()`? I also think we should use the pythonic way of getting uid ([getters and setters](https://stackoverflow.com/questions/2627002/whats-the-pythonic-way-to-use-getters-and-setters)) so that we can throw an error if stream.uid is None.